labcontrol/db/process.py

```diff
@@ -3244,18 +3248,40 @@ def generate_sample_sheet(self):
     def generate_prep_information(self):
         """Generates prep information
 
+        Returns
+        -------
+        dict labcontrol.db.study.Study: str
```

Comment:
Will do!

Reply:
Issue created as https://github.com/jdereus/labman/issues/494. Verifying the return type.

Reply:
In brief, the return type is a dictionary, with both an integer key (the study id and n prep sheet ids), and a string key (the 'Control' label). The values for each of the keys are TSV files in string format. I do recall this output from earlier work correcting one of the columns in the output.

It can get a little confusing, because the data dictionary is not only what's returned, but is also main working variable throughout the entire process. Moreover, the method encapsulates the generation of both variants of the TSV output, and performs them both inside a loop. It would be nice to refactor the loop out and have encapsulated variants for control and sample data. Moreover, we should just return the DataFrames, and have the conversion to TSV occur downstream in the loop, or wherever.

At any rate, I can change the documentation to reflect what's current, and see if I can't refactor a little.

Reply:
@charles-cowart  Do not refactor too much--recall that we will be (perhaps temporarily) putting ALL prep output into a SINGLE sheet for next week's code freeze, so we will no longer have multiple entries in this dictionary.  Suggest holding off on work on this until after May 16 when (hopefully) stakeholders reach an agreement with Rob about how to handle controls.